src/components/permissions/rowLevelPolicyHelpers.ts

```diff
@@ -167,18 +176,33 @@ const compileRowLevelPolicy = (
   usingCondition: string,
   condition: string
 ) => {
+  // Create function for permission so that row level policies are checked independently
+  // SECURITY DEFINER is used to make sure permission check does not take into account existing permissions
+  // i.e. without this function, if we have something like this "restrict review by review_assignment where user is reviewer"
+  // and then we have another restriction of "review_assignment where user is assigner" in the former policy check the latter check will also run
+  const functionName = `private.policy_check_${permissionAbbreviation}_${tableName}`
```

Comment:
Bit of a naming inconsistency here. `permissionAbbreviation` is called both `permissionPolicyAbbreviation` and `policyAbbreviation` above. I think having the word `policy` in there in important. `policyAbbreviation` works for me, but go with the longer `permissionPolicyAbbreviation` if you want to be 100% clear at the expense of brevity.